mindee/parsing/common/inference.py

```diff
@@ -0,0 +1,72 @@
+from typing import Dict, Generic, List, Optional, TypeVar
+
+from mindee.parsing.common.page import TypePage
+from mindee.parsing.common.prediction import TypePrediction
+from mindee.parsing.common.string_dict import StringDict
+
+
+class Inference(Generic[TypePrediction, TypePage]):
+    """Base Inference class for all predictions."""
+
+    product_name: str
+    """Name of the product as sent by the API."""
+    product_version: str
+    """Version of the product as sent by the API."""
+    endpoint_name: Optional[str]
+    """Name of the endpoint for OTS APIs"""
+    endpoint_version: Optional[str]
+    """Version of the endpoint for OTS APIs"""
```

Comment:
I could see these attributes set as class decorators.

They are not really part of the object, just a useful way for us to associate the needed info to call the API.

Reply:
I don't mind, but having these as simple static attributes felt like the most "pan-sdk" solution to me.

Reply:
Java and .NET use decorators ... but we can use this solution for "lesser" programming languages :trollface:

Reply:
Sorry I don't speak boilerplate